lib/vinyl-stream.js

```diff
@@ -35,6 +35,9 @@ module.exports = function createVinylStream(options) {
 				path: base + '/' + object.Key
 			});
 
+	  if(!file.contents)
+		  file.contents =  new Buffer("");
+
 		// Set the length on things like streams which are used
 		// by some frameworks.
 		if (!file.contents.length) {
```

Comment:
The `file` object itself won't be null, but `file.contents` will (which is why the error is occurring) and `file.isNull()` does indeed check the condition `file.contents === null`.

> then you may want to set the content length so that you can make use of the file size.

For things with content I think this makes sense. If there is no content to access then `file.contents` should be null (as in the metadata case). If you need the `content-length` value of the metadata then it can be accessed via `file.stat.size` which mimics the `vinyl-fs` way of doing things.

 > you need to make the contents into a buffer, or Vinyl will throw exceptions.

`vinyl` is quite happy with null contents; directories, for example, are a kind of entry with no contents. Certainly any exceptional case is on the developer's hands and not on vinyl's.